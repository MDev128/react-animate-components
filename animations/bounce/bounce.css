@keyframes bounce {
  from {
    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);
  }

  0% {
    opacity: 0;
    transform: scale3d(.3, .3, .3);
  }

  20% {
    transform: scale3d(1.1, 1.1, 1.1);
  }

  40% {
    transform: scale3d(.9, .9, .9);
  }

  60% {
    opacity: 1;
    transform: scale3d(1.1, 1.1, 1.1);
  }

  80% {
    transform: scale3d(.97, .97, .97);
  }

  to {
    opacity: 1;
    transform: scale3d(1, 1, 1);
  }
}

.bounce {
  animation-name: bounce
}
