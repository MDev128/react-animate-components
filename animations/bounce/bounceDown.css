@keyframes bounceDown {
  from {
    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);
  }

  0% {
    transform: translate3d(0, -2000px, 0);
  }

  60% {
    transform: translate3d(0, 25px, 0);
  }

  75% {
    transform: translate3d(0, -15px, 0);
  }

  90% {
    transform: translate3d(0, 5px, 0);
  }

  to {
    transform: translate3d(0, 0, 0);
  }
}

.bounceDown {
  animation-name: bounceDown
}
