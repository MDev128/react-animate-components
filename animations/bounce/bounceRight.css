@keyframes bounceRight {
  from, 60% 75% 90%, to {
    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);
  }

  0% {
    opacity: 0;
    transform: translate3d(2000px, 0, 0);
  }

  60% {
    opacity: 1;
    transform: translate3d(-25px, 0, 0);
  }

  75% {
    transform: translate3d(10px, 0, 0);
  }

  90% {
    transform: translate3d(-5px, 0, 0);
  }

  to {
    transform: none;
  }
}

.bounceRight {
  animation-name: bounceRight
}
