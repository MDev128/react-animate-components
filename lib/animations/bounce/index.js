'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _styledBounce = require('./styledBounce');

var _HOC = require('../HOC');

var _HOC2 = _interopRequireDefault(_HOC);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Bounce animation components
var bounceObj = {
	Bounce: (0, _HOC2.default)('Bounce', _styledBounce.bounce),
	BounceDown: (0, _HOC2.default)('BounceDown', _styledBounce.bounceDown),
	BounceUp: (0, _HOC2.default)('BounceUp', _styledBounce.bounceUp),
	BounceLeft: (0, _HOC2.default)('BounceLeft', _styledBounce.bounceLeft),
	BounceRight: (0, _HOC2.default)('BounceRight', _styledBounce.bounceRight)
};

exports.default = bounceObj;