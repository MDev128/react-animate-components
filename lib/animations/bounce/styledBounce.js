'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.bounceRight = exports.bounceLeft = exports.bounceUp = exports.bounceDown = exports.bounce = undefined;

var _templateObject = _taggedTemplateLiteral(['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: scale3d(.3, .3, .3);\n  }\n\n  20% {\n    transform: scale3d(1.1, 1.1, 1.1);\n  }\n\n  40% {\n    transform: scale3d(.9, .9, .9);\n  }\n\n  60% {\n    opacity: 1;\n    transform: scale3d(1.1, 1.1, 1.1);\n  }\n\n  80% {\n    transform: scale3d(.97, .97, .97);\n  }\n\n  to {\n    opacity: 1;\n    transform: scale3d(1, 1, 1);\n  }\n'], ['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: scale3d(.3, .3, .3);\n  }\n\n  20% {\n    transform: scale3d(1.1, 1.1, 1.1);\n  }\n\n  40% {\n    transform: scale3d(.9, .9, .9);\n  }\n\n  60% {\n    opacity: 1;\n    transform: scale3d(1.1, 1.1, 1.1);\n  }\n\n  80% {\n    transform: scale3d(.97, .97, .97);\n  }\n\n  to {\n    opacity: 1;\n    transform: scale3d(1, 1, 1);\n  }\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    transform: translate3d(0, -2000px, 0);\n  }\n\n  60% {\n    transform: translate3d(0, 25px, 0);\n  }\n\n  75% {\n    transform: translate3d(0, -15px, 0);\n  }\n\n  90% {\n    transform: translate3d(0, 5px, 0);\n  }\n\n  to {\n    transform: translate3d(0, 0, 0);\n  }\n'], ['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    transform: translate3d(0, -2000px, 0);\n  }\n\n  60% {\n    transform: translate3d(0, 25px, 0);\n  }\n\n  75% {\n    transform: translate3d(0, -15px, 0);\n  }\n\n  90% {\n    transform: translate3d(0, 5px, 0);\n  }\n\n  to {\n    transform: translate3d(0, 0, 0);\n  }\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    transform: translate3d(0, 2000px, 0);\n  }\n\n  60% {\n    transform: translate3d(0, -25px, 0);\n  }\n\n  75% {\n    transform: translate3d(0, 10px, 0);\n  }\n\n  90% {\n    transform: translate3d(0, -5px, 0);\n  }\n\n  to {\n    transform: translate3d(0, 0, 0);\n  } \n'], ['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    transform: translate3d(0, 2000px, 0);\n  }\n\n  60% {\n    transform: translate3d(0, -25px, 0);\n  }\n\n  75% {\n    transform: translate3d(0, 10px, 0);\n  }\n\n  90% {\n    transform: translate3d(0, -5px, 0);\n  }\n\n  to {\n    transform: translate3d(0, 0, 0);\n  } \n']),
    _templateObject4 = _taggedTemplateLiteral(['\n  from, 60% 75% 90%, to {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: translate3d(-2000px, 0, 0);\n  }\n\n  60% {\n    opacity: 1;\n    transform: translate3d(25px, 0, 0);\n  }\n\n  75% {\n    transform: translate3d(-10px, 0, 0);\n  }\n\n  90% {\n    transform: translate3d(5px, 0, 0);\n  }\n\n  to {\n    transform: none;\n  } \n'], ['\n  from, 60% 75% 90%, to {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: translate3d(-2000px, 0, 0);\n  }\n\n  60% {\n    opacity: 1;\n    transform: translate3d(25px, 0, 0);\n  }\n\n  75% {\n    transform: translate3d(-10px, 0, 0);\n  }\n\n  90% {\n    transform: translate3d(5px, 0, 0);\n  }\n\n  to {\n    transform: none;\n  } \n']),
    _templateObject5 = _taggedTemplateLiteral(['\n  from, 60% 75% 90%, to {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: translate3d(2000px, 0, 0);\n  }\n\n  60% {\n    opacity: 1;\n    transform: translate3d(-25px, 0, 0);\n  }\n\n  75% {\n    transform: translate3d(10px, 0, 0);\n  }\n\n  90% {\n    transform: translate3d(-5px, 0, 0);\n  }\n\n  to {\n    transform: none;\n  } \n'], ['\n  from, 60% 75% 90%, to {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: translate3d(2000px, 0, 0);\n  }\n\n  60% {\n    opacity: 1;\n    transform: translate3d(-25px, 0, 0);\n  }\n\n  75% {\n    transform: translate3d(10px, 0, 0);\n  }\n\n  90% {\n    transform: translate3d(-5px, 0, 0);\n  }\n\n  to {\n    transform: none;\n  } \n']);

var _styledComponents = require('styled-components');

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var bounce = (0, _styledComponents.keyframes)(_templateObject);

var bounceDown = (0, _styledComponents.keyframes)(_templateObject2);

var bounceUp = (0, _styledComponents.keyframes)(_templateObject3);

var bounceLeft = (0, _styledComponents.keyframes)(_templateObject4);

var bounceRight = (0, _styledComponents.keyframes)(_templateObject5);

exports.bounce = bounce;
exports.bounceDown = bounceDown;
exports.bounceUp = bounceUp;
exports.bounceLeft = bounceLeft;
exports.bounceRight = bounceRight;