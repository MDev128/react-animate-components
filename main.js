/**
 * @author Nitin Tulswani <tulswani19@gmail.com>
 */

// Animation Component.
import bounceObj from './dist/animations/bounce/index';
import fadeObj from './dist/animations/fade/index';
import flipObj from './dist/animations/flip/index';
import lightObj from './dist/animations/lightspeed/index';
import rotateObj from './dist/animations/rotate/index';
import specialObj from './dist/animations/special';
import slideObj from './dist/animations/slide/index';
import PObj from './dist/animations/perspective/index';
import bingoObj from './dist/animations/Bingo/index';
import expanseObj from './dist/animations/expanse/index';

// Exposing keyframes for explicit use.
import {
	puffmeout,
	puffmein,
	vanishout,
	vanishin
} from './dist/animations/Bingo/styledBingo';

import {
  bounce,
  bounceDown,
  bounceUp,
  bounceLeft,
  bounceRight,
} from './dist/animations/bounce/styledBounce';

import { expanseUp, expanseDown, expanseLeft, expanseRight } from './dist/animations/expanse/styledExpanse';

import {
  down,
  downBig,
  up,
  upBig,
  left,
  right,
  leftBig,
  rightBig,
  fadeIn,
} from './dist/animations/fade/styledFade';

import {
	flip,
	flipX,
	flipY,
} from './dist/animations/flip/styledFlip';

import {
	lightIn,
	lightOut,
} from './dist/animations/lightspeed/styledLight';

import { perspectiveDown, perspectiveUp, perspectiveRight, perspectiveLeft } from './dist/animations/perspective/styledPerspective';

import { slideUp, slideDown, slideLeft, slideRight } from './dist/animations/slide/styledSlide';

import {
  rotateIn,
  rotateLeft,
  rotateRight,
  rotateUpRight,
  rotateUpLeft,
} from './dist/animations/rotate/styledRotate';

import {
  flash,
  rollOut,
  rollIn,
  rubber,
  swing,
  zoom,
  hinge,
  pulse,
  expandUp,
  entrance,
  hatch,
  starWars
} from './dist/animations/styledSpecial';

const { Bounce, BounceUp, BounceRight, BounceLeft, BounceDown } = bounceObj;

const {
  FadeIn,
  FadeInUp,
  FadeInRight,
  FadeInLeft,
  FadeInDown,
  FadeInUpBig,
  FadeInRightBig,
  FadeInLeftBig,
} = fadeObj;

const { Flip, FlipX, FlipY } = flipObj;

const { LightOut, LightIn } = lightObj;

const {
  RotateIn,
  RotateRight,
  RotateLeft,
  RotateUpRight,
  RotateUpLeft,
} = rotateObj;

const {
  Flash,
  RollOut,
  RollIn,
  Rubber,
  Swing,
  Zoom,
  Hinge,
  Pulse,
  ExpandUp,
  Entrance,
  Hatch,
  StarWars
} = specialObj;

const { SlideUp, SlideRight, SlideLeft, SlideDown } = slideObj;

const { PDown, PUp, PRight, PLeft } = PObj;

const { PuffOut, PuffIn, VanishOut, VanishIn } = bingoObj;

const { ExpanseUp, ExpanseRight, ExpanseDown, ExpanseLeft } = expanseObj;

/** keyframes */
export {
  puffmeout,
  puffmein,
  vanishout,
  vanishin,
  bounce,
  bounceDown,
  bounceUp,
  bounceLeft,
  bounceRight,
  expanseUp,
  expanseDown,
  expanseLeft,
  expanseRight,
  down,
  downBig,
  up,
  upBig,
  left,
  right,
  leftBig,
  rightBig,
  fadeIn,
  flip,
	flipX,
	flipY,
  lightIn,
	lightOut,
  perspectiveDown,
  perspectiveUp,
  perspectiveRight,
  perspectiveLeft,
  slideUp,
  slideDown,
  slideLeft,
  slideRight,
  rotateIn,
  rotateLeft,
  rotateRight,
  rotateUpRight,
  rotateUpLeft,
  flash,
  rollOut,
  rollIn,
  rubber,
  swing,
  zoom,
  hinge,
  pulse,
  expandUp,
  entrance,
  hatch,
  starWars
}

/** Animation components */
export {
  Bounce,
  BounceUp,
  BounceRight,
  BounceLeft,
  BounceDown,
  FadeIn,
  FadeInUp,
  FadeInRight,
  FadeInLeft,
  FadeInDown,
  FadeInUpBig,
  FadeInLeftBig,
  FadeInRightBig,
  Flip,
  FlipX,
  FlipY,
  LightOut,
  LightIn,
  RotateIn,
  RotateRight,
  RotateLeft,
  RotateUpRight,
  RotateUpLeft,
  Flash,
  RollOut,
  RollIn,
  Rubber,
  Swing,
  Zoom,
  Hinge,
  Pulse,
  SlideUp,
  SlideDown,
  SlideLeft,
  SlideRight,
  StarWars,
  ExpandUp,
  Entrance,
  Hatch,
  PDown,
  PUp,
  PRight,
  PLeft,
  PuffIn,
  PuffOut,
  VanishIn,
  VanishOut,
  ExpanseLeft,
  ExpanseRight,
  ExpanseDown,
  ExpanseUp
};
