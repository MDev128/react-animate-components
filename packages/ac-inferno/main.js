module.exports = require('./dist/ac-inferno.min.js');
