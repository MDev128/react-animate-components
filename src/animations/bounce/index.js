// @flow

import Bouncing from './styledBounce';
import HOC from '../../containers/HOC';
import type { Components } from '../../../types';

const bounceObj: Components = {
  Bounce: HOC('Bounce', Bouncing.bounce),
  BounceDown: HOC('BounceDown', Bouncing.bounceDown),
  BounceUp: HOC('BounceUp', Bouncing.bounceUp),
  BounceLeft: HOC('BounceLeft', Bouncing.bounceLeft),
  BounceRight: HOC('BounceRight', Bouncing.bounceRight),
};

export default bounceObj;
