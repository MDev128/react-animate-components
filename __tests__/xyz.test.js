import React from "react";
import renderer from "react-test-renderer";
import { keyframes } from "styled-components";
import { shallow, mount } from "enzyme";

import { Bounce } from "../main";
import * as All from "../main";

import { bounce } from "../src/animations/bounce/styledBounce";
import bounceObj from "../src/animations/bounce/index";

const bounceWrapper = (
  <Bounce>
    <h1>Hello World</h1>
  </Bounce>
);

const bounceBlock = (
  <Bounce block>
    <h1>Hello World!</h1>
  </Bounce>
);

const bounceDefaultProps = {
  children: <h1>Hello World</h1>,
  duration: "1s",
  timingFunction: "ease",
  delay: "0s",
  direction: "normal",
  iterations: "1",
  backfaceVisible: "visible",
  fillMode: "none",
  playState: "running"
};

it("keyframes should be string", () => {
  expect(typeof bounce).toBe("string");
});

it("Animation component renders it children", () => {
  const tree = renderer.create(bounceWrapper).toJSON();
  expect(tree).toMatchSnapshot();
});
it("Animation component renders it children with props", () => {
  const tree = renderer
    .create(
      <Bounce duration="2s" delay="4s" direction="reverse">
        <h1>Hello World!</h1>
      </Bounce>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Animation component default props", () => {
  const wrapper = shallow(bounceWrapper);
  expect(wrapper.instance().props).toEqual(bounceDefaultProps);
});

it("Animation component renders it children with `block` prop", () => {
  const tree = renderer.create(bounceBlock).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Animation component calls componentDidMount lifecycle method", () => {
  const wrapper = shallow(bounceWrapper);
  wrapper.instance().componentDidMount();
});

it("Animation component updates the styles when the it mounts", () => {
  const wrapper = shallow(bounceWrapper);
  wrapper.instance().componentDidMount();
  expect(wrapper.state('styles')).toEqual({
    animation: `
            bGJZDZ 1s ease 0s 1 normal
            none running
          `,
    backfaceVisibility: "visible" });
});

it("Animation component calls renderRootWithBlock method when passed `block` prop", () => {
  const wrapper = shallow(bounceBlock);
  wrapper.instance().renderRootWithBlock();
});

it("calls renderRootWithInline method (by default when no display props are passed)", () => {
  const wrapper = shallow(bounceWrapper);
  wrapper.instance().renderRootWithInline();
});

it("calls shouldComponentUpdate lifecycle method for avoiding re-render of the component", () => {
  const wrapper = shallow(bounceWrapper);
	wrapper.setProps({ duration: '1s' });
  // Avoids re-render (returns true)
  wrapper.instance().shouldComponentUpdate();
});

it("each component should export an object containing different animation components", () => {
  expect(typeof bounceObj).toBe("object");
  expect(Object.keys(bounceObj).length).not.toBe(0);
});

it("exports all the animation components and keyframes", () => {
  expect(typeof All).toBe("object");
  expect(Object.keys(All).length).toBe(51);
});
