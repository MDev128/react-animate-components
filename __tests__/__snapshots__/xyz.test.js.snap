// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animation component renders it children 1`] = `
<span
  style={
    Object {
      "animation": "
                bGJZDZ 1s ease 0s 1 normal
                none running
              ",
      "backfaceVisibility": "visible",
      "display": "inline-block",
    }
  }
>
  <h1>
    Hello World
  </h1>
</span>
`;

exports[`Animation component renders it children with \`block\` prop 1`] = `
<div
  style={
    Object {
      "animation": "
                bGJZDZ 1s ease 0s 1 normal
                none running
              ",
      "backfaceVisibility": "visible",
      "display": "block",
    }
  }
>
  <h1>
    Hello World!
  </h1>
</div>
`;

exports[`Animation component renders it children with props 1`] = `
<span
  style={
    Object {
      "animation": "
                bGJZDZ 2s ease 4s 1 reverse
                none running
              ",
      "backfaceVisibility": "visible",
      "display": "inline-block",
    }
  }
>
  <h1>
    Hello World!
  </h1>
</span>
`;
